File: CodeGenDotNet 159059-1372 (a)/CSharp/Tools/CodeGenerationSupport/DatabaseSupport.cs
// ====================================================================
//  Copyright ©2004, Kathleen Dollard, All Rights Reserved.
// ====================================================================
//   I'm distributing this code so you'll be able to use it to see code
//   generation in action and I hope it will be useful and you'll enjoy 
//   using it. This code is provided "AS IS" without warranty, either 
//   expressed or implied, including implied warranties of merchantability 
//   and/or fitness for a particular purpose. 
// ====================================================================
//  Summary: Utility files for database actions

using System;
using System.Text.RegularExpressions;
using System.Diagnostics;

namespace KADGen.CodeGenerationSupport
{
	/// <summary>
	/// 
	/// </summary>
	public class DatabaseSupport
	{
		#region Class level declarations - empty
		#endregion

		#region Constructors - empty
		#endregion

		#region Public Methods and Properties
		public static string GetConnectionString( string serverName, string databaseName )
		{
			return "workstation id=" + serverName + ";packet size=4096;integrated security=SSPI;data source=" + serverName + ";persist security info=False;initial catalog=" + databaseName;
		}

		public static void CreateStoredProcFromFile(	string fileName,
			string serverName,
			string databaseName,
			string spName,
			string executeUser )
		{
			System.IO.StreamReader streamReader = new System.IO.StreamReader( fileName );
			System.Data.SqlClient.SqlCommand cmd = new System.Data.SqlClient.SqlCommand();
			System.Data.SqlClient.SqlTransaction transaction = null;
			//string[] SQLStatements;
			cmd.Connection = new System.Data.SqlClient.SqlConnection( CodeGenerationSupport.DatabaseSupport.GetConnectionString( serverName, databaseName ) );
			cmd.CommandText = "sp_help '" + spName + "'";
			try
			{
				cmd.Connection.Open();
				transaction = cmd.Connection.BeginTransaction();
				cmd.Transaction = transaction;
				string[] statements = Regex.Split( streamReader.ReadToEnd(), "\\sgo\\s", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.IgnoreCase );
				foreach( string statement in statements )
				{
					cmd.CommandText = statement;
					cmd.ExecuteNonQuery();
				}
				transaction.Commit();
			}
			catch( System.Exception ex )
			{
				System.Diagnostics.Debug.WriteLine( ex );
				transaction.Rollback();
				throw;
			}
			finally
			{
				try
				{
					cmd.Connection.Close();
				}
				catch
				{
				}
			}
		}
		#endregion
	}
}

File: CodeGenDotNet 159059-1372 (a)/VBCode/Tools/CodeGenerationSupport/DatabaseSupport.vb
' ====================================================================
'  Copyright ©2004, Kathleen Dollard, All Rights Reserved.
' ====================================================================
'   I'm distributing this code so you'll be able to use it to see code
'   generation in action and I hope it will be useful and you'll enjoy 
'   using it. This code is provided "AS IS" without warranty, either 
'   expressed or implied, including implied warranties of merchantability 
'   and/or fitness for a particular purpose. 
' ====================================================================
'  Summary: Utility files for database actions

Option Strict On
Option Explicit On

Imports System
Imports System.Text.RegularExpressions
Imports System.Diagnostics

'! Class Summary: 

Public Class DatabaseSupport
   
#Region "Class level declarations - empty"
#End Region

#Region "Constructors - empty"
#End Region

#Region "Public Methods and Properties"
   Public Shared Function GetConnectionString( _
               ByVal serverName As String, _
               ByVal databaseName As String) _
               As String
      Return "workstation id=" & serverName & ";packet size=4096;integrated security=SSPI;data source=" & serverName & ";persist security info=False;initial catalog=" & databaseName
   End Function

   Public Shared Sub CreateStoredProcFromFile( _
            ByVal fileName As String, _
            ByVal serverName As String, _
            ByVal databaseName As String, _
            ByVal spName As String, _
            ByVal executeUser As String)
      Dim streamReader As New IO.StreamReader(fileName)
      Dim cmd As New Data.SqlClient.SqlCommand
      Dim transaction As Data.SqlClient.SqlTransaction
      Dim SQLStatements() As String
      cmd.Connection = New Data.SqlClient.SqlConnection(CodeGenerationSupport.DatabaseSupport.GetConnectionString(serverName, databaseName))
      cmd.CommandText = "sp_help '" & spName & "'"
      Try
         cmd.Connection.Open()
         transaction = cmd.Connection.BeginTransaction
         cmd.Transaction = transaction
         Dim statements() As String = Regex.Split(streamReader.ReadToEnd, "\sgo\s", RegexOptions.Compiled Or RegexOptions.CultureInvariant Or RegexOptions.IgnoreCase)
         For Each statement As String In statements
            cmd.CommandText = statement
            cmd.ExecuteNonQuery()
         Next
         transaction.Commit()
      Catch ex As System.Exception
         Debug.WriteLine(ex)
         transaction.Rollback()
         Throw
      Finally
         Try
            cmd.Connection.Close()
         Catch
         End Try
      End Try
   End Sub

   'Public Shared Sub CreateStoredProcFromFile( _
   '            ByVal fileName As String, _
   '            ByVal serverName As String, _
   '            ByVal databaseName As String, _
   '            ByVal spName As String, _
   '            ByVal executeUser As String)
   '   Dim streamReader As New IO.StreamReader(fileName)
   '   Dim cmd As New Data.SqlClient.SqlCommand
   '   cmd.Connection = New Data.SqlClient.SqlConnection(CodeGenerationSupport.DatabaseSupport.GetConnectionString(serverName, databaseName))
   '   cmd.CommandText = "sp_help '" & spName & "'"
   '   Try
   '      cmd.Connection.Open()
   '      Try
   '         cmd.ExecuteNonQuery()
   '         cmd.CommandText = "drop procedure [dbo].[" & spName & "]"
   '         cmd.ExecuteNonQuery()
   '      Catch ex As Exception
   '         ' Benign, the stored proc doesn't exist
   '      End Try
   '      cmd.CommandText = streamReader.ReadToEnd
   '      cmd.ExecuteNonQuery()
   '      cmd.CommandText = "GRANT  EXECUTE  ON [dbo].[" & spName & "]  TO [" & executeUser & "]"
   '      cmd.ExecuteNonQuery()
   '   Finally
   '      Try
   '         cmd.Connection.Close()
   '      Catch
   '      End Try
   '   End Try
   'End Sub

#End Region

#Region "Protected and Friend Methods and Properties - empty"
#End Region

#Region "Private Methods and Properties - empty"
#End Region

End Class
